### PR TITLE
feat: add completed quests table

### PR DESCRIPTION
**Pull request by lauti7**
This PR:
- rename `abandoned_quests` to `abandoned_quest_instances`
- adds new table to record completed quest instances

### CODE REVIEW

File: crates/db/src/lib.rs [hunk 1]
```diff
@@ -342,22 +342,49 @@ impl QuestsDatabase for Database {
...
+        let id = Uuid::new_v4().to_string();
+        let query = sqlx::query(
+            "INSERT INTO completed_quest_instances (id, quest_instance_id) VALUES ($1, $2)",
+        )
+        .bind(parse_str_to_uuid(&id)?)
+        .bind(parse_str_to_uuid(quest_instance_id)?);
         query
             .execute(&self.pool)
             .await
             .map_err(|err| DBError::DeactivateQuestFailed(Box::new(err)))
```
Comment (guidota) [inline crates/db/src/lib.rs:367]: I think we should add a new error for this case

File: crates/db/src/lib.rs [hunk 2]
```diff
@@ -342,22 +342,49 @@ impl QuestsDatabase for Database {
...
     }
 
+    async fn is_completed_instance(&self, quest_instance_id: &str) -> DBResult<bool> {
+        let quest_instance_exists: bool = sqlx::query_scalar(
+            "SELECT EXISTS (SELECT 1 FROM completed_quest_instances WHERE quest_instance_id = $1)",
+        )
+        .bind(parse_str_to_uuid(quest_instance_id)?)
+        .fetch_one(&self.pool)
+        .await
+        .map_err(|err| DBError::GetActiveQuestInstanceFailed(Box::new(err)))?;
```
Comment (guidota) [inline crates/db/src/lib.rs:378]: Same here

File: crates/system/src/event_processing.rs [hunk 3]
```diff
@@ -161,6 +161,8 @@ impl EventProcessor {
             .await?;
 
         if quest_state.is_completed() {
+            debug!("Processing event > recording instance as completed");
+            self.database.complete_quest(quest_instance_id).await?;
```
Comment (guidota) [inline crates/system/src/event_processing.rs:165]: I would do this after the reward is given or try to handle the error

**Review by guidota — COMMENTED** [review]
The code looks good. Left some comments.

File: crates/db/src/lib.rs (hunk 2 above)
Comment (lauti7) [reply crates/db/src/lib.rs:378]: sorry copypasting

**Review by guidota — APPROVED** [review]

**Pull request merged**